[Teo Sin Yee]  iP

tsinyee opened the pull request:
### **DukePro**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download the `.jar` file from [here](https://github.com/tsinyee/ip/releases/tag/A-Jar).
2. Run it using `java -jar ip.jar`.
3. Add your tasks.
4. Let it manage your tasks for you! 😉

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Built on very easy to understand code. Example snippet of the `main` function:

```
    public static void main (String[] args) {
        Duke duke = new Duke();
        duke.start();
    }
```

Commit:
Add Gradle support

tsinyee committed:
Level-1

tsinyee committed:
Level-2: Add, List

tsinyee committed:
Level-3: Mark as Done

tsinyee committed:
Level-4: ToDo, Event, Deadline

tsinyee committed:
A-TextUiTesting: Automated Text UI Testing

tsinyee committed:
Level-5 & Level-6: Handle Errors & Delete

tsinyee committed:
Added documentation and more error handling

tsinyee committed:
A-TextUiTesting

tsinyee committed:
Completed Level-7: Save

tsinyee committed:
Completed Level-8: Dates and Times

tsinyee committed:
Merge branch 'branch-Level-8'

tsinyee committed:
Refactored code and added more OOP to code

tsinyee committed:
Divided classes into packages

tsinyee committed:
Added subclasses for Command

tsinyee committed:
Update Storage.java

tsinyee committed:
Added subclasses for DukeException

tsinyee committed:
Merge remote-tracking branch 'origin/master'

tsinyee committed:
Add JUnit tests

tsinyee committed:
Add more subclasses for DukeException

tsinyee committed:
Package the App as a JAR file

tsinyee committed:
Add find functionality

tsinyee committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-Level-9

tsinyee committed:
Add JavaDoc

tsinyee committed:
Merge branch 'branch-A-JavaDoc'

tsinyee committed:
Add coding standards

tsinyee committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/util/Ui.java

tsinyee committed:
Add find functionality

tsinyee committed:
Merge branch 'branch-Level-9'

tsinyee committed:
Add JavaDoc comments

tsinyee committed:
Add JavaDoc comments

tsinyee committed:
Merge branch 'branch-A-JavaDoc'

tsinyee committed:
Integrate Gradle

tsinyee committed:
Merge branch 'branch-A-Gradle'

tsinyee committed:
Update intro message

tsinyee committed:
Follow coding standards

tsinyee committed:
Implement Checkstyle

tsinyee committed:
Improve code to follow coding standards

tsinyee committed:
Merge branch 'branch-A-Checkstyle'

chowRuiYan commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,25 @@
+ *
+ * @author Teo Sin Yee
+ * @version CS2103T AY21/22 Semester 1
+ */
```
I like how you separate all the different commands into individual files, making it very neat and easy to read. 👍

chowRuiYan commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,201 @@
+    public Command parseRawInput(String input) throws DukeException {
+        String[] splitInput = input.split("\\s+");
+        String commandWord = splitInput[0];
+        switch (commandWord) {
```
Perhaps you could make the input case-insensitive. 
```suggestion
        switch (commandWord.toLowerCase()) {
```
So instead of throwing a `DukeNoSuchCommandException()` when, for example, *"List"* is given as input, it can still work as expected.

chowRuiYan reviewed (commented):
I think your code is very well written and I don't have much to comment on. Great job! 😄

tsinyee committed:
Add GUI for Duke

tsinyee committed:
Merge branch 'branch-Level-10'

tsinyee committed:
Update build.gradle

chetwinlow commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    /**
+     * String representation of a deadline.
```
Maybe the comment could be changed to "Returns a string ..."?

chetwinlow commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,201 @@
+    }
+
+    /**
+     * Extract the date for deadline and event commands.
```
Probably careless but you probably meant"Extracts..." ?

chetwinlow commented on src/main/java/duke/util/TaskHandler.java:
```diff
@@ -0,0 +1,165 @@
+     *
+     * @return The number of incomplete tasks.
+     */
+    private int getUndoneTasksCount() {
```
Might seem a bit wordy? May I suggest something like "getNumUndone"?

chetwinlow commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,124 @@
+        }
+    }
+
+    /** Creates a new file to store data **/
```
I do believe that the comment should end with "*/"?

chetwinlow reviewed (commented):
Overall code is clean and well formatted!

tsinyee committed:
Improve GUI and add JavaDocs

tsinyee committed:
Add assertions

There are some preconditions in the code that must be satisfied before the program can run successfully.

Adding assertions will help us define assumptions about the program state so that the runtime can verify them. An assertion failure indicates a possible bug and the sooner the execution stops, the safer it is.

Let's,
* add assertions to test assumptions about the program

tsinyee committed:
Improve code quality

There are several methods that do not follow the code quality standards as specifed in the textbook.

Improving the code quality will significantly increase readability and
understandability of the code. This helps ourselves and other developers
who wish to use the code in the future.

Let's,
* refactor the code to increase readability

tsinyee committed:
Merge pull request #1 from tsinyee/branch-A-Assertions

Add assertions

tsinyee committed:
Merge branch 'master' into branch-A-CodeQuality

tsinyee committed:
Merge pull request #2 from tsinyee/branch-A-CodeQuality

Improve code quality

tsinyee committed:
Add sort functionality

The task list is ordered according to users' sequence of adding tasks.

Adding a sort function would greatly improve user experience as users can now view the task list sorted in chronological order.

As a step towards enhancing user experience, let's implement a sort function for the task list.

tsinyee committed:
Merge branch 'branch-A-CodeQuality'

tsinyee committed:
Improve GUI

tsinyee committed:
Update JavaDocs

tsinyee committed:
Update JUnit Tests

tsinyee committed:
Fix bug where event/deadline name is not extracted correctly

tsinyee committed:
Set theme jekyll-theme-cayman

tsinyee committed:
Update messages shown to user

tsinyee committed:
Update message shown to user when task list is printed

tsinyee committed:
Move Launcher class to duke package

tsinyee committed:
Add User Guide

tsinyee committed:
Merge remote-tracking branch 'origin/master'

tsinyee committed:
Update User Guide

tsinyee committed:
Update build.gradle

tsinyee committed:
Update User Guide

tsinyee committed:
Update User Guide

tsinyee committed:
Update Java Docs
